storages/backends/gcloud.py

```diff
@@ -0,0 +1,233 @@
+from tempfile import SpooledTemporaryFile
+
+from django.core.exceptions import ImproperlyConfigured
+from django.core.files.base import File
+from django.core.files.storage import Storage
+from django.utils.deconstruct import deconstructible
+from django.utils.encoding import force_bytes, force_text, smart_str
+from django.utils import timezone
+from storages.utils import clean_name, safe_join, setting
+
+try:
+    from google.cloud.storage.client import Client
+    from google.cloud.storage.blob import Blob
+    from google.cloud.exceptions import NotFound
+except ImportError:
+    raise ImproperlyConfigured("Could not load Google Cloud Storage bindings.\n"
+                               "See https://github.com/GoogleCloudPlatform/gcloud-python")
+
+
+class GoogleCloudFile(File):
+    def __init__(self, name, mode, storage):
+        self.name = name
+        self._mode = mode
+        self._storage = storage
+        self.blob = storage.bucket.get_blob(name)
+        if not self.blob and 'w' in mode:
+            self.blob = Blob(self.name, storage.bucket)
+        self._file = None
+        self._is_dirty = False
+
+    @property
+    def size(self):
+        return self.blob.size
+
+    def _get_file(self):
+        if self._file is None:
+            self._file = SpooledTemporaryFile(
+                max_size=self._storage.max_memory_size,
+                suffix=".GSStorageFile",
+                dir=setting("FILE_UPLOAD_TEMP_DIR", None)
+            )
+            if 'r' in self._mode:
+                self._is_dirty = False
+                self.blob.download_to_file(self._file)
+                self._file.seek(0)
+        return self._file
+
+    def _set_file(self, value):
+        self._file = value
+
+    file = property(_get_file, _set_file)
+
+    def read(self, num_bytes=None):
+        if 'r' not in self._mode:
+            raise AttributeError("File was not opened in read mode.")
+        if num_bytes is not None:
+            return super(GoogleCloudFile, self).read(num_bytes)
+        else:
+            return super(GoogleCloudFile, self).read()
```

Comment:
I'd be surprised too but [Django specifies `num_bytes` as the kwarg](https://docs.djangoproject.com/en/dev/ref/files/file/#django.core.files.File.read) so it would be reasonable for someone to do that, and really confusing to debug if they did.

Reading the Django docs a bit more closely I noticed that the text gives the name of the argument as `size`, so there's an inconsistency there. I think the best way forward is to submit a PR to fix the Django docs, wait for a version with the fix to be released, _then_ change this argument (here and in other backends).

In any case I've made the `if` change you suggested because your suggestion makes it a bit clearer.